src/words/TokTok.jl

```diff
@@ -67,46 +61,44 @@ const rules_atoms = Tuple(Iterators.flatten([
         EN_EM_DASHES,
         PROB_SINGLE_QUOTES,
         OPEN_PUNCT,
-        CLOSE_PUNCT    
-    ])
+        CLOSE_PUNCT]))
 
-const rules_replaces = vcat(
-        NON_BREAKING,
-        AMPERCENT,
-        TAB,
-        PIPE,
-        STUPID_QUOTES_1,
-        STUPID_QUOTES_2,
-        ONE_SPACE
-    )
+const rules_replaces = Tuple(Iterators.flatten([
+        [AMPERCENT],
+        [TAB],
+        [PIPE],
+        [STUPID_QUOTES_1],
+        [STUPID_QUOTES_2]]))
 
-const rules_splitting = vcat(LSTRIP, RSTRIP)
 
 """
     totok_tokenize(instring::AstractString)
-This tokenizer is a simple, general tokenizer, where the input has one sentence per line; thus only final period is tokenized. 
+This tokenizer is a simple, general tokenizer, where the input has one sentence per line; thus only final period is tokenized.
 Tok-tok has been tested on and gives reasonably good results for English, Persian, Russian, Czech, French, German, Vietnamese,
 Tajik, and a few others.
 """
 function toktok_tokenize(instring::AbstractString)
     ts = TokenBuffer(instring)
     isempty(ts.input) && return ts.tokens
- 
-    flag = handle_final_periods(ts)
-    
-    while !isdone(ts) && ts.idx <= flag
-        url_handler1(ts) ||   # these url handlers have priority over others
+
+    effective_end = handle_final_periods(ts)
+
+    while !isdone(ts) && ts.idx <= effective_end
+	if string(ts.input[ts.idx]) == NON_BREAKING[1]
+		ts.input[ts.idx] = NON_BREAKING[2]
+	end
+
+        url_handler4(ts) ||   # these url handlers have priority over others
```

Comment:
Found that this should be the correct priority when suing this approach instead of the regex version.